### PR TITLE
[tempo-distributed] Add gateway component to tempo-distributed

### PR DESCRIPTION
**Pull request by daryamorgan**
Hello!

In my company we need to push jaeger traces to tempo, located in a different cluster, this is possible via gateway ingress.

I added gateway component which proxies API requests to appropriate tempo components. The configuration is pretty much the same as in loki-distributed gateway component. The changes are backwards compatible, gateway is disabled by default.

I also found an issue that suggests adding ingress for tempo components https://github.com/grafana/helm-charts/issues/551.

Please have a look if it's fine.
Thank you for the great product!

### CODE REVIEW

**Review by annanay25 — COMMENTED** [review]
Thanks for your contribution @daryamorgan! This is something that I have been wanting to do this quarter and glad that it happened sooner.

Left a few minor comments.

File: charts/tempo-distributed/values.yaml [hunk 1]
```diff
@@ -526,3 +531,209 @@ serviceMonitor:
...
+        access_log   /dev/stderr  main;
+        sendfile     on;
+        tcp_nopush   on;
+        resolver {{ .Values.global.dnsService }}.{{ .Values.global.dnsNamespace }}.svc.{{ .Values.global.clusterDomain }};
+
+        {{- with .Values.gateway.nginxConfig.httpSnippet }}
+        {{ . | nindent 2 }}
+        {{- end }}
+
+        server {
```
[the remark this answers is not in the document]
Comment (daryamorgan) [reply charts/tempo-distributed/values.yaml:686]: @annanay25 Do you suggest to add one endpoint `/config` to direct to some specific service or to add several endpoints, one for each service that supports this API e.g. `{service}/config`?

File: charts/tempo-distributed/values.yaml [hunk 2]
```diff
@@ -526,3 +531,209 @@ serviceMonitor:
...
+          auth_basic_user_file /etc/nginx/secrets/.htpasswd;
+          {{- end }}
+
+          location = / {
+            return 200 'OK';
+            auth_basic off;
+          }
+
+          location /jaeger/ {
+            proxy_pass       http://{{ include "tempo.distributorFullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}:14268/;
```
[the remark this answers is not in the document]
Comment (daryamorgan) [reply charts/tempo-distributed/values.yaml]: @annanay25 I corrected the endpoint for jaeger, to expose only /jaeger/api/traces on gateway.
A also specified endpoint for otlp and zipking http traces according to the documentation.
Please take a look if it's ok

File: charts/tempo-distributed/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (annanay25) [reply charts/tempo-distributed/values.yaml:686]: I think that would be a bit much, let's merge this one as is.

**Review by annanay25 — DISMISSED** [review]
Think there's some linting changes to go, but otherwise LGTM, thank you very much!

**Review by annanay25 — APPROVED** [review]

**Comment by annanay25** [comment]
Thanks @daryamorgan!

**Pull request merged**